lms/djangoapps/courseware/grades.py

```diff
@@ -727,3 +737,14 @@ def iterate_grades_for(course_or_id, students, keep_raw_scores=False):
                     exc.message
                 )
                 yield student, {}, exc.message
+
+
+def _get_mock_request(student):
+    """
+    Make a fake request because grading code expects to be able to look at
+    the request. We have to attach the correct user to the request before
+    grading that student.
+    """
+    request = RequestFactory().get('/')
```

Comment:
@ormsbee would there be any concern here with this change - which is, instead of making one `RequestFactory` object during this code, we now make one object _per student_? I don't know much about Python's garbage collecting or memory allocations but I am concerned about the creation of potentially hundreds of thousands of objects within the loop of one function, particularly the grading functions which are notoriously slow.